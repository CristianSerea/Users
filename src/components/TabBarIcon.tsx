import * as React from 'react';
import {Image, ImageSourcePropType, StyleSheet} from 'react-native';
import useColorsTheme from '../constants/colors';

type Props = {
  source: ImageSourcePropType;
};

const TabBarIcon = ({source}: Props) => {
  const {colors} = useColorsTheme();

  return (
    <Image
      source={source}
      style={[styles.image, {tintColor: colors.mainColor}]}
      testID="TabBarIcon"
    />
  );
};

const styles = StyleSheet.create({
  image: {
    width: 22,
    height: 22,
  },
});

export default TabBarIcon;
