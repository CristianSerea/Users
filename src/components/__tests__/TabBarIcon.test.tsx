import React from 'react';
import {render} from '@testing-library/react-native';
import TabBarIcon from '../TabBarIcon';

describe('TabBarIcon tests', () => {
  it('Test tab bar icon rendering', () => {
    const source = require('../../assets/home.png');
    const {getByTestId, toJSON} = render(<TabBarIcon source={source} />);
    const image = getByTestId('TabBarIcon');

    expect(image).toBeTruthy();
    expect(image.props.source).toEqual(source);

    expect(toJSON()).toMatchSnapshot();
  });
});
