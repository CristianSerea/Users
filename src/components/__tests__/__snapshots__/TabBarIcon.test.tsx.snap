// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarIcon tests Renders tab bar icon 1`] = `
<Image
  source={
    {
      "testUri": "../../../src/assets/home.png",
    }
  }
  style={
    [
      {
        "height": 22,
        "width": 22,
      },
      {
        "tintColor": "#008000",
      },
    ]
  }
  testID="TabBarIcon"
/>
`;

exports[`TabBarIcon tests Test tab bar icon rendering 1`] = `
<Image
  source={
    {
      "testUri": "../../../src/assets/home.png",
    }
  }
  style={
    [
      {
        "height": 22,
        "width": 22,
      },
      {
        "tintColor": "#008000",
      },
    ]
  }
  testID="TabBarIcon"
/>
`;
